cmd/autobumper/main.go

```diff
@@ -2,6 +2,286 @@
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/robots/pr-creator/updater"
+)
+
+const (
+	prowPrefix      = "gcr.io/k8s-prow/"
+	testImagePrefix = "gcr.io/k8s-testimages/"
+	prowRepo        = "https://github.com/kubernetes/test-infra"
+	testImageRepo   = prowRepo
+	githubOrg       = "openshift"
+	githubRepo      = "release"
+)
+
+func cloneRepo(dir string) error {
+	logrus.Infof("Changing working directory to %s...", dir)
+	if err := os.Chdir(dir); err != nil {
+		return fmt.Errorf("failed to chdir to (%s): %v", dir, err)
+	}
+	repoURL := fmt.Sprintf("https://github.com/%s/%s", githubOrg, githubRepo)
+	logrus.Infof("cloning repo %s...", repoURL)
+	cmd := exec.Command("git", "clone", "--depth", "1", repoURL)
+	_, err := cmd.Output()
+	return err
+}
+
+func cdToRootDir(dir string) error {
+	logrus.Infof("Changing working directory to %s...", dir)
+	return os.Chdir(dir)
+}
+
+type options struct {
+	githubLogin string
+	githubToken string
+	gitName     string
+	gitEmail    string
+}
+
+func parseOptions() options {
+	var o options
+	flag.StringVar(&o.githubLogin, "github-login", "", "The GitHub username to use.")
+	flag.StringVar(&o.githubToken, "github-token", "", "The path to the GitHub token file.")
+	flag.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	flag.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+	flag.Parse()
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login is mandatory")
+	}
+	if o.githubToken == "" {
+		return fmt.Errorf("--github-token is mandatory")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	return nil
+}
+
+func call(cmd string, args ...string) error {
+	c := exec.Command(cmd, args...)
+	c.Stdout = os.Stdout
+	c.Stderr = os.Stderr
+	return c.Run()
+}
+
+func updatePR(gc github.Client, org, repo, title, body, matchTitle, source, branch string) error {
+	logrus.Info("Creating PR...")
+	n, err := updater.UpdatePR(org, repo, title, body, matchTitle, gc)
+	if err != nil {
+		return fmt.Errorf("failed to update %d: %v", n, err)
+	}
+	if n == nil {
+		pr, err := gc.CreatePullRequest(org, repo, title, body, source, branch, true)
+		if err != nil {
+			return fmt.Errorf("failed to create PR: %v", err)
+		}
+		n = &pr
+	}
+
+	logrus.Infof("PR %s/%s#%d will merge %s into %s: %s", org, repo, *n, source, branch, title)
+	return nil
+}
+
+func updateReferences() (map[string]string, error) {
+	logrus.Info("Bumping image references...")
+	filter := regexp.MustCompile(prowPrefix + "|" + testImagePrefix)
+
+	err := filepath.Walk("./cluster/ci/config/prow/openshift/", func(path string, info os.FileInfo, err error) error {
+		if strings.HasSuffix(path, ".yaml") {
```

Comment:
let us allow also `.yml` files.

Reply:
No `*.yml` yet in our target folders.

```
$ find . -name '*.yml'
./projects/prometheus/alertmanager.yml
./cluster/ci/monitoring/mixins/vendor/kubernetes-mixin/.circleci/config.yml
```

I would rename them if any
https://stackoverflow.com/questions/22268952/what-is-the-difference-between-yaml-and-yml-extension/22268973